plugin/advisor/pg/advisor_index_no_duplicate_column.go

```diff
@@ -0,0 +1,161 @@
+
+	for _, stmt := range stmtList {
+		checker.text = stmt.Text()
+		checker.line = stmt.LastLine()
+		ast.Walk(checker, stmt)
+	}
+
+	if len(checker.adviceList) == 0 {
+		checker.adviceList = append(checker.adviceList, advisor.Advice{
+			Status:  advisor.Success,
+			Code:    advisor.Ok,
+			Title:   "OK",
+			Content: "",
+		})
+	}
+
+	return checker.adviceList, nil
+}
+
+type indexNoDuplicateColumnChecker struct {
+	adviceList []advisor.Advice
+	level      advisor.Status
+	title      string
+	text       string
+	line       int
+}
+
+type duplicateColumn struct {
+	table          string
+	index          string
+	column         string
+	constraintType string
+	line           int
+}
+
+func (checker *indexNoDuplicateColumnChecker) Visit(node ast.Node) ast.Visitor {
+	var columnList []duplicateColumn
+	switch node := node.(type) {
+	case *ast.CreateTableStmt:
+		for _, constraint := range node.ConstraintList {
+			switch constraint.Type {
+			case ast.ConstraintTypePrimary,
+				ast.ConstraintTypeForeign,
+				ast.ConstraintTypePrimaryUsingIndex,
+				ast.ConstraintTypeUnique,
+				ast.ConstraintTypeUniqueUsingIndex:
+				if column, duplicate := hasDuplicateColumn(constraint.KeyList); duplicate {
+					columnList = append(columnList, duplicateColumn{
+						table:          node.Name.Name,
+						index:          constraint.Name,
+						column:         column,
+						constraintType: contraintsTypeToString(constraint.Type),
+						line:           checker.line,
+					})
+				}
+			}
+		}
+	case *ast.CreateIndexStmt:
+		if column, duplicate := hasDuplicateColumn(node.Index.GetKeyNameList()); duplicate {
+			columnList = append(columnList, duplicateColumn{
+				table:          node.Index.Table.Name,
+				index:          node.Index.Name,
+				column:         column,
+				constraintType: "INDEX",
+				line:           checker.line,
+			})
+		}
+	case *ast.AlterTableStmt:
+		for _, item := range node.AlterItemList {
+			switch cmd := item.(type) {
+			case *ast.AddConstraintStmt:
+				switch cmd.Constraint.Type {
+				case ast.ConstraintTypePrimary,
+					ast.ConstraintTypeForeign,
+					ast.ConstraintTypePrimaryUsingIndex,
+					ast.ConstraintTypeUnique,
+					ast.ConstraintTypeUniqueUsingIndex:
+					if column, duplicate := hasDuplicateColumn(cmd.Constraint.KeyList); duplicate {
+						columnList = append(columnList, duplicateColumn{
+							table:          cmd.Table.Name,
+							index:          cmd.Constraint.Name,
+							column:         column,
+							constraintType: contraintsTypeToString(cmd.Constraint.Type),
+							line:           checker.line,
+						})
+					}
+				}
+			default:
+				continue
+			}
+		}
+	}
+
+	for _, column := range columnList {
+		checker.adviceList = append(checker.adviceList, advisor.Advice{
+			Status:  checker.level,
+			Code:    advisor.DuplicateColumnInIndex,
+			Title:   checker.title,
+			Content: fmt.Sprintf("%s `%s` has duplicate column `%s`.`%s`", column.constraintType, column.index, column.table, column.column),
```

Comment:
How about using the `"` instead of the `. In PostgreSQL, we use double quotes to mark the identifier. 
The backquote is the MySQL style.